### ISSUE TITLE
Classic CS team

### ISSUE THREAD

**Issue by Xirzz** [kolton/d2bot-with-kolbot#460]
I think classic need more love the current Diablo scripts aren't adapted for classic at all. can we do something about it?

**Comment by noah-** [comment]
Have you tried ClassicChaosAssistant.js with the normal diablo script?

**Comment by Xirzz** [comment]
Wait no I never tried, I just noticed that this script is in the bots folder but how can I activate it? there is nothing in the sorc script that says anything about that.

Also are you Noah that made a special classic script posted on blizzhacker forum??

**Comment by DiaLover** [comment]
noah- is god

**Comment by Wiener93** [comment]
Im using AutoCs and AutoTaxi. dunno where i got it from i think from blizzhackers. i modified it tho and im doing 1/8 cs rund with a sorc that taxis and waits in town for pala to clear seal, then taxis next seal. easy hardcore classic solo fast dia. if you want i can pastebin it.

**Comment by Xirzz** [comment]
that sounds pretty good for a first script to use I would appreciate if you could pastebin it Wiener93 

My ultimate goal would be to have taxi sorc + hammerdin + barb singer (just casting the fear warcry constantly) + bo barb of course, do you think I can make it work with your script?

**Comment by noah-** [comment]
Yeah, I would've mentioned AutoCS, but blizzhackers is down and I don't have a copy of it anymore.

**Comment by Wiener93** [comment]
AutoTaxi: https://pastebin.com/biS11wuk
AutoCS: https://pastebin.com/En58iGyg
Installation Directions: https://pastebin.com/7Vsh0kCs

Noah´s Description ;)

Special thanks to Rayne for testing and validating these scripts for all characters available in classic Diablo II. All testing was handled purely by him, I never even had to run the script once!

What is AutoChaos?
AutoChaos is a two script system AutoTaxi and AutoCS.
AutoTaxi is a taxi style script, it creates a portal at each seal and does not kill monsters in between seals.
AutoCS is a follow script, it will automatically determine a taxi leader and follow it through the various seals in CS.

How is AutoChaos different from existing scripts?
AutoChaos does not use any form of in game messaging or out of game communication scheme. Think of AutoChaos as if a human player was joining a public cs game. You don't need to be told who the leader is, it's usually the highest level sorceress or a character that already has a tp up in cs. You don't need to be told when to go in the tp, when you see the leader making a new tp, you can make a good assumption that the next seal is up. AutoChaos uses these same rules to lead/follow cs runs. AutoChaos leverages in game data about portal GID and player locations to determine which seal you are at and what to do. 

AutoChaos is meant to be a proof of concept: how to manage team games without knowing any info about the team before hand. The AutoCS script should be a compatible follower to any taxi based cs, whether it be human run or otherwise. It should be able to join any game and adapt to the current situation. Ideally with AutoChaos, people that don't have the ability to run full teams could run 1 character having them join public games that contain the name "Chaos" or "CS" or any of the other names for cs games and be able to form a team on the fly. 

Features: 
AutoChaos is not meant to be the fastest most optimized cs script. It is however meant to be very reliable and consistent. These scripts are provided as is and without any guarantees. Also, I will not be adding any more feature updates, only bug fixes. Please do not make feature requests unless they are related to reliability/consistency of the script. 

The AutoTaxi script is meant to be run by a sorceress, the AutoCS script can be run with any type of character. Builds that work really well with this setup include: cold sorceress, ce/ranged necro, javazon, hammerdin, ww barb (or bo barb) 
Other setups may require changes to the attack routines.


> Maybe you´ll include it in the Master Branch someday? I could get a new clean install, implement it and share the changes if you like Noah.

**Comment by noah-** [comment]
I don't think it was ever extensively tested which is why I never added it to kolbot. Feel free to post your feedback and I will consider it.

**Comment by Xirzz** [comment]
btw in the Taxi script is the sorc attacking or she just do a tp and go town or back to star? I mean the original script. It is important as in HC you dont want the sorc to tp and stop in big pack of monsters it would make her chicken right away

**Comment by 3594302** [comment]
BH DOWN.

**Comment by Xirzz** [comment]
I'm trying to run the scripts but I keep getting this;  Error in AutoCS (loader.js #132) Invalid script function name 

The sorc is waiting at river WP so it seems to be ok with her, it's just an issue with the ones  running AutoCS
also another question is now the BO is happening at river right? so I have to turn the first BO scripts to false?

**Comment by Wiener93** [comment]
Sry i edited the post. link to AutoCS should work now. ye noah i understand that. theres still work to do to make it worth for a public script. i´ll take a look at if its worth it to fix it 100%. i know its not flawless. and @Xirzz sorc fights with the other chars with this normal AutoTaxi script.

you can take a look at my AutoTaxi script here: https://pastebin.com/XuNpu8Mg (wont work properly with standard autocs i think) if you want.

**Comment by Xirzz** [comment]
Wiener I tried everything but the barb doesnt seem to want to BO at river what should I do?

**Comment by Wiener93** [comment]
its noahs script :/ i dont use a barb in cs. i use the normal BattleOrders.js for Bo barb except i use river of flames wp instead of Catacombs lvl 2 wp.

**Comment by Xirzz** [comment]
Pather.useWaypoint(35, true); // catacombs

What number is it for River of flames?

**Comment by Wiener93** [comment]
Pather.useWaypoint(107, true); // catacombs  // GEÄNDERT ZU AKT 4 CATA IST WP NR 35

german :D

**Comment by Xirzz** [comment]
thank you it works! but it's a very dangerous script. 

You think we can modify this script and make it more efficient with some ideas I have? One of the things that would be very important is the sorc to ONLY make tps and never fight at seals she head to a seal make the tp at the boss location/pop the seal/go back to star and wait XX seconds/ if there is a monster attacking her she will randomly tp at the star/then head to next seal when boss is dead

this would be the most important thing that made it very safe for the sorc not to die and the rest is a big kill squad with hammerdin and barb that do the fear warcry so the monsters run around and they can focus on killing the seal boss

Do you see that like something possible to do?

**Comment by Wiener93** [comment]
use this for your sorc https://pastebin.com/XuNpu8Mg you´ll have to add the pala as leader in your char config tho. might confuse you to setup the invite/ accept options for party since when leader is set the sorc will only accept party from pala. so barb need to be set to accept too.

**Comment by Xirzz** [comment]
Ok I'm gonna try this one out.

Any idea on how I can make the AutoCS script wait longer before leaving a game if members of the squad are not detected? sometime it's a bit too fast and after a crash it needs time to reconnect and get back to the game (freaking eternal Checking Versions while wanting to connect...)

**Comment by Xirzz** [comment]
I get this with your script:

[Error(73)] File (kolbot/libs/bots/autotaxi.js:318) SyntaxError: missing } after function body
Failted to include script: Autotaxi

What went wrong??

**Comment by Wiener93** [comment]
try again. i updated the link to my custom Autotaxi.js. maybe its because i commented out the exp shrine getter.

**Comment by Xirzz** [comment]
yesssss it's working she goes in town after every seal popped this is very nice thank you Wiener93 I'm gonna do some test and see how it goes. she tries to always get a shrine when Diablo is being done so I'd like to remove that

**Comment by Wiener93** [comment]
Try these
Taxi: https://pastebin.com/Dy52mdm3
AutoCS: https://pastebin.com/Zn8RVD1L
Sorc and Others shouldnt quit anymore. Im bad at variables, so im hardcoding everything. Sorc stays for Diablo now instead of getting exp shrine. You can make your pala also get the exp shrine the sorc makes tp at ofc. like i said my goal was to solo pala exp.

remember all changes i make needs testing. im not responsible for anything.

For barb change 

> failTimer = 5000,

in your BattleOrders.js to whatever you want (5000 is changed by me already).

**Comment by Xirzz** [comment]
thank you wiener <3 starting the tests tomorrow after work. you rock.

**Comment by Xirzz** [comment]
Hi wiener this error is back again with the last taxi script;

Error(133)] File (kolbot/libs/bots/autotaxi.js:196) SyntaxError: syntax error
Failed to include script: Autotaxi

**Comment by Xirzz** [comment]
Error(133)] File (kolbot/libs/bots/autotaxi.js:196) SyntaxError: syntax error
Failed to include script: Autotaxi 

again :/

**Comment by Wiener93** [comment]
sorry mate has been to messy. i started from the default autotaxi again. now it should work: https://pastebin.com/xHn379mN

tomorrow ill take a further look into the AutoCS.js

**Comment by Wiener93** [comment]
sry mate busy tonight. ill let you know in the next few days. they should work kinda okay now? i´ll just be finetuning a bit. add the option to choose if sorc will fight or wait in town, optimize AutoCS etc.

**Comment by Xirzz** [comment]
woooow you rock dude thanks so much, with upcoming ladder reset it's gonna be amazing. I put these to test and come back for the feedbacks!! 👍

**Comment by Xirzz** [comment]
how do you make the barb bo at river without the BO.js script?? when using your script he just stays in town

also I'm getting this on the sorc whenever I ask her to kill diablo after finding the xp shrine for pala.

Error in ClassicTeamTaxi (classicteamtaxi.js line 217): Config.AutoCS is undefined 

Also she will sometime miss the tp back to diablo if the paladin take it before her.

**Comment by toosaa** [comment]
hello, is there way run other other scripts after diablo is done? all players just quit game and go to next.

**Comment by sez55** [comment]
@Wiener93 

hey i have a problem with these scripts, Problem is that the sorce goes to viz makes tp open seals but goes diretctly to seis make a tp open seals and go directy to inf and make tp open seals and goes directy in middle of cs (Star) .... but why ? maybe you can help   sorry for my english

**Comment by ATragicEnding** [comment]
Is there any way to implement what appears to be random Battle Orders from these scripts into the regular Chaos Helper script? I'd rather clear whole CS but the helper script apparently only BO every X minutes thus having the sorc chicken out because of the lack of Battle Order :-/

**Comment by ClassicBaby** [comment]
this script working? same problem as @chojo31 sorcs sit at WP and everyone sits in town, will sometimes start and tp visior kill then afk till chicken 

thanks for help

**Comment by 77920** [comment]
Not working... anyone got an update?

**Comment by anime143** [comment]
bump, iso help. 
my setup is based on xpac style but i want it to be classic only. 

i followed as instructed but doesn't seem to work unless i switch it to xpac config like mf helper/ diablo help.

**Comment by noah-** [comment]
Here is updated autochaos
https://gist.github.com/noah-/2685fbeccc72fd595bbe89116aea272e
